android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt

```diff
@@ -178,6 +183,14 @@ fun ButtonProperties.interpolateStatus(computedValuesMap: Map<String, Any>): Ser
   else ServiceStatus.UPCOMING
 }
 
+@Composable
+fun ButtonProperties.interpolateBackgroundColor(computedValuesMap: Map<String, Any>): Color {
+  val interpolated = this.backgroundColor?.interpolate(computedValuesMap)
+  return if (!interpolated.isNullOrEmpty()) {
+    return interpolated.parseColor()
+  } else Color.Unspecified
+}
```

Comment:
what if combine interpolation for any (or all the ones we want to) ButtonProperties into a single step, any thoughts on how we can do that?

relatedly, wondering if these interpolation functions (or a more general one) should live in ButtonProperties and not here.

relatedly, a little uneasy that we're combining interpolation and setting a default (with the else) into single step, that's a bit broader than only interpolation b/c the name is mislead -- I see this pattern was introduced before so this is not a critique of this specific addition, it's following that pattern

Reply:
Yeah sure, I'll move the interpolations to one function in the button properties responsible for handling all the interpolations.